### DISCUSSION TITLE
Custom system message selection and searXNG url do not appear to "load" when you close and open browser.

### DISCUSSION THREAD

**Discussion by fhasbrookATW** [mamei16/LLM_Web_search#14] [Q&A]
I was wondering if I'm doing something wrong.  I have added a SearXNG URL and also selected the default websearch_gpt4_prompt.  I hit save settings and I can see the SearXNG update into the settings.jason file, but do not see the custom system message save there.  However upon closing the browser and going back in, the settings seem to have reverted back to default on the display even though the SearXNG URL is still in the settings.jason.   Is this just a display error and and the SearXNG URL is being used?  I would also like to know how to set the gpt4_prompt as the default if this is the case.  Thank you in advance.

**Comment by mamei16** [comment]
Hi, thank you for asking this! The UI elements returning to default values is indeed just a display bug and definitely not intended. You can actually trigger it just by refreshing the page, without even needing to open and close the browser. I guess I just never did that and hence never noticed it lol. I just now fixed this in commit 704414c53636deddbe99942d38f5804812ba77a3*
Custom system messages are not saved in the `settings.json` file. Instead, each of them is saved in a separate file inside the `system_prompts` sub-folder in the extension's main folder. Setting one as default wasn't a option before, but I added this option now, since I also think it could be useful: 00a45a62d74ab7a54cadda3bb4bf38d71b00a764

*note that the some of the elements in the main web UI will still not reload correctly after refreshing, such as the Generation Preset and instruction template drop-downs.

**Comment by fhasbrookATW** [comment]
Thank you those changes are working perfectly!   Might I suggest you create a requirements.txt  and add it to the repo?

duckduckgo-search==4.1.0
beautifulsoup4==4.12.2
langchain==0.0.352
unstructured==0.6.6
rank_bm25==0.2.2
sentence-transformers==2.2.2

I have this running in Kubernetes with the below env variables and it builds the dependencies as it brings the pods up. But I had to create the requirements.txt for it to do so.

           - name: BUILD_EXTENSIONS_LIVE
              value: LLM_Web_search

**Reply by mamei16** [reply]
Good to hear! 
I chose to go with a conda-based installation only, since the [faiss-cpu pip package](https://github.com/kyamagu/faiss-wheels) is unofficial and I was not sure whether it could be relied upon. But I'll add a requirements.txt while still keeping the installation instructions the same, so that those who don't need installation instructions in the first place can freely choose between conda and pip installations. Thanks for that suggestion.
 
If you don't already use pip install caching and/or have multiple pods running the the web UI + extensions, I think you should consider building a docker image with the dependencies already installed, so you don't have to install them each time a pod is started.
